### DISCUSSION TITLE
I need help sending custom information to session.user object.

### DISCUSSION THREAD

**Discussion by JuanSpada** [nextauthjs/next-auth#7252] [Help] [answered]
### Question 💬

Hello, Im tryng to add custom data to the session.user but it doesnt work. The odd thing is that if I send this:
`const user = {
    id: '1',
    name: "J Smith",
    email: "jsmith@example.com",
  };`
Then I receive that information on my page.jsx using useSession and print it in the console:
`const { data: session } = useSession();
console.log("in form session: ", session);`

But the problem is when I try to add custom information to the user:
`const user = {
  userId: res.data.data.login.userId,
  companyId: res.data.data.login.companyId,
};`

It printing empty the user object, without the custom information, only works when I use id, name, and email.

Someone knows how to fix this?
Thanks

### How to reproduce ☕️
```
import NextAuth from "next-auth";
import CredentialsProvider from "next-auth/providers/credentials";
import axios from "axios";

const handler = NextAuth({
  providers: [
    CredentialsProvider({
      name: "Credentials",
      credentials: {
        email: {
          label: "Email",
          type: "email",
          placeholder: "Enter your email",
        },
        password: { label: "Password", type: "password" },
      },
      async authorize(credentials, req) {
        const res = await axios({
          method: "POST",
          url: `${process.env.NEXT_PUBLIC_AUTH_URL}/api`,
          data: {
            query: `
              query Login($email: String!, $password: String!) {
                login(email: $email, password: $password) {
                  userId
                  token
                  tokenExpiration
                  companyId
                }
              }
            `,
            variables: {
              email: credentials.email,
              password: credentials.password,
            },
          },
        });
        // Validation
        if (res.data.errors) {
          throw new Error(res.data.errors[0].message);
        }
        const user = {
          userId: res.data.data.login.userId,
          companyId: res.data.data.login.companyId,
        };
        if (user) {
          return user;
        }
        // // Return null if user data could not be retrieved
        return null;
      },
    }),
  ],
  session: {
    strategy: "jwt",
  },
  pages: {
    signIn: "/login",
    signOut: "/logout",
  },
  secret: process.env.NEXTAUTH_SECRET,
});

export { handler as GET, handler as POST };

```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by rei-iku** [comment]
you need custom information using this library? start making plans to move to a own built solution

**Reply by balazsorban44** [reply]
Not sure about the sentiment here, what the OP asks is fully possible https://github.com/nextauthjs/next-auth/discussions/7252#discussioncomment-5615425

If you have constructive feedback, please let us know via opening an issue/bug report. :pray:

**Comment by balazsorban44** [comment, marked as answer]
https://next-auth.js.org/getting-started/example#extensibility

Check out the `jwt` callback, which will receive the `user` from the Credentials provider, and the `session` callback that will expose the data you need in the client (eg. `useSession`)

By default we don't expose everything for security reasons.

**Comment by JuanSpada** [comment]
Thank you @balazsorban44 that fix my issue.

**Comment by JuanSpada** [comment]
I run into another issue now, When I refresh the page the session it's undefined, but If then I go to another route and back the session has the token.
So its only happening when I refresh the page. Did you know how can I fix this? @balazsorban44
